augur/ancestral.py

```diff
@@ -191,36 +247,46 @@ def run(args):
     # explicitly or by default) and the user has not explicitly requested that
     # we keep them.
     infer_ambiguous = args.infer_ambiguous and not args.keep_ambiguous
-
-    tt = ancestral_sequence_inference(tree=T, aln=aln, ref=ref, marginal=args.inference,
-                                      fill_overhangs = not(args.keep_overhangs),
-                                      infer_tips = infer_ambiguous)
-
-    character_map = {}
-    for x in tt.gtr.profile_map:
-        if tt.gtr.profile_map[x].sum()==tt.gtr.n_states:
-            # TreeTime treats all characters that are not valid IUPAC nucleotide chars as fully ambiguous
-            # To clean up auspice output, we map all those to 'N'
-            character_map[x] = 'N'
-        else:
-            character_map[x] = x
-
-    anc_seqs.update(collect_mutations_and_sequences(tt, full_sequences=not is_vcf,
-                          infer_tips=infer_ambiguous, character_map=character_map))
-    # add reference sequence to json structure. This is the sequence with
-    # respect to which mutations on the tree are defined.
-    if is_vcf:
-        anc_seqs['reference'] = {"nuc":compress_seq['reference']}
-    else:
-        root_seq = tt.sequence(T.root, as_string=False)
-        if anc_seqs.get("mask") is not None:
-            root_seq[anc_seqs['mask']] = tt.gtr.ambiguous
-        anc_seqs['reference'] = {"nuc": ''.join(root_seq)}
+    full_sequences = not is_vcf
+    nuc_result = run_ancestral(T, aln, root_sequence=str(ref.seq) if ref else None, is_vcf=is_vcf, fill_overhangs=not args.keep_overhangs,
+                               full_sequences=full_sequences, marginal=args.inference, infer_ambiguous=infer_ambiguous, alphabet='nuc')
+    anc_seqs = nuc_result['mutations']
+    anc_seqs['reference'] = {'nuc': nuc_result['root_seq']}
 
     if anc_seqs.get("mask") is not None:
         anc_seqs["mask"] = "".join(['1' if x else '0' for x in anc_seqs["mask"]])
 
-    anc_seqs['annotations'] = {'nuc': {'start': 1, 'end': len(anc_seqs['reference']['nuc']), 'strand': '+'}}
+    anc_seqs['annotations'] = {'nuc': {'start': 1, 'end': len(anc_seqs['reference']['nuc']),
+                                       'strand': '+', 'type': 'source'}}
```

Comment:
What's the purpose of `type: source`?

This change is the reason for the failing tests. The test can be [easily changed](https://github.com/nextstrain/augur/commit/9a53e7d8a35e929c69425a54c0f04f5b06ed87f3) to pass again, but I want to understand the reasoning first.

Reply:
`source` is the name of the sequence with respect to which annotations in genbank files are made. If you look at [this file for example](https://www.ncbi.nlm.nih.gov/nuccore/KJ627695) you see a `source` field at the beginning of the annotation block. In GFF these are called landmarks. In augur translate we assign this also for the feature type `source`:

https://github.com/nextstrain/augur/blob/master/augur/translate.py#L395-L400

I added this here just in analogy to `translate`, I don't think auspice requires it.

Reply:
While Auspice doesn't use this, the `"type"` key (with `string` value) is specified [in our node-data JSON annotations schema](https://github.com/nextstrain/augur/blob/9ef471129fc048f596952af5cbee4cd87feac20b/augur/data/schema-annotations.json#L15-L18) and [exported (auspice) JSON schema](https://github.com/nextstrain/augur/blob/9ef471129fc048f596952af5cbee4cd87feac20b/augur/data/schema-export-v2.json#L68-L72) so tests shouldn't fail.